### DISCUSSION TITLE
Could be Laravel run totally in memory?

### DISCUSSION THREAD

**Discussion by ProteanCode** [laravel/framework#39409] [Ideas] [answered]
In a free time I am thinking if Laravel could be preloaded fully from memory.

This would require that everything that Laravel caches to be "regeneratable" on demand, ie. if you have 100 blade files then you would need to be able to create its cached version without visiting the view by navigation through app (or you have to create some awkward crawling scripts). This should be doable as cached view does not depend on execution flow of blade statements, whenever user is admin or not the cached version of the blade file at ```layouts.admin.header``` heading will stay the same.

Cache for config files already works that way afaik.

Alternatively we could get a way to cache views at some ephemeral redis store? I think that this is not that obvious

After all of that, it would require PHP to preload the code responsible. Ideally the whole /app directory
https://www.php.net/manual/en/opcache.preloading.php

The .env files could be also loaded from a Redis store (actaully env is also cache'able as a config command part, but redis is more microservice oriented feature on the stack and allows us to give a little more security)

At this point there would be close to none i/o disk hits, ight?

To fulfill some of the possible preassumes:
Yes, the deployment of such app would require the PHP engine and a small downtime on a single hosted node

**Comment by dennisprudlo** [comment, marked as answer]
Do you mean like octane? https://laravel.com/docs/8.x/octane
It keeps your app in memory

**Reply by ProteanCode** [reply]
> Do you mean like octane? https://laravel.com/docs/8.x/octane It keeps your app in memory

Interesting.

The only downside to that is that I think the everyday Laravel developer is not aware how it works, it basically feels too unadapted, or edgy.

Nobody wants to see a production server returning 502 / 503 because of the edgy php coroutine mechanism instead of simply scaling up containers amount to temporary solve the problem.

This is especially truth if the whole network is running in Rounded Robin mesh where no server is being hit twice in a row so usage of coroutines is limited unless REALLY heavy load.
